### ISSUE TITLE
SNOW-1348115: Add support for column tags

### ISSUE THREAD

**Issue by ogr003** [Snowflake-Labs/terraform-provider-snowflake#1074]
**Is your feature request related to a problem? Please describe.**

Snowflake tables support tags, but columns do not. This stops us from managing all tags within Terraform.

**Describe the solution you'd like**

The ability to implement tags for columns in the same way as for tables:

```hcl
resource "snowflake_tag" "test_tag" {
  database       = "TEST_DB"
  schema         = "TEST_SCHEMA"
  name           = "TEST_TAG"
  allowed_values = ["TABLE", "COLUMN"]
}

resource "snowflake_table" "test_table" {
  database = snowflake_schema.schema.database
  schema   = snowflake_schema.schema.name
  name     = "TEST_TABLE"

  tag {
    name     = snowflake_tag.test_tag.name
    database = snowflake_tag.test_tag.database
    schema   = snowflake_tag.test_tag.schema
    value    = "TABLE"
  }

  column {
    name     = "data"
    type     = "text"
    nullable = false

    tag {
      name     = snowflake_tag.test_tag.name
      database = snowflake_tag.test_tag.database
      schema   = snowflake_tag.test_tag.schema
      value    = "COLUMN"
    }
  }
}
```

**Comment by sfc-gh-kumaurya** [comment]
Sure, we do need this feature in our provider. I'll start working on this feature request.

**Comment by sfc-gh-jmichalak** [comment]
Hi @ogr003  👋

We've just released v0.100.0 ([release](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.100.0), [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0990--v01000)) in which we reworked tag associations. Tags can be assigned to columns in the `tag_association` resource.
Please upgrade to this version and let us know if you have any issues.

**Comment by sfc-gh-dszmolka** [comment]
closing this out as it should be fixed now as my colleague mentioned. Do let us know if you still see issues with v1.
